econml/dr/_drlearner.py

```diff
@@ -158,13 +158,21 @@ def score(self, Y, T, X=None, W=None, *, nuisances, sample_weight=None, sample_v
             X = self._featurizer.transform(X)
         Y_pred, = nuisances
         if self._multitask_model_final:
-            return np.mean(np.average((Y_pred[..., 1:] - Y_pred[..., [0]] - self.model_cate.predict(X))**2,
-                                      weights=sample_weight, axis=0))
+            Y_pred_diff = Y_pred[..., 1:] - Y_pred[..., [0]]
+            cate_pred = self.model_cate.predict(X)
+            if self.d_y:
+                cate_pred = cate_pred[:, np.newaxis, :]
+            return np.mean(np.average((Y_pred_diff - cate_pred)**2, weights=sample_weight, axis=0))
+
         else:
-            return np.mean([np.average((Y_pred[..., t] - Y_pred[..., 0] -
-                                        self.models_cate[t - 1].predict(X))**2,
-                                       weights=sample_weight, axis=0)
-                            for t in np.arange(1, Y_pred.shape[-1])])
+            scores = []
+            for t in np.arange(1, Y_pred.shape[-1]):
+                # since we only allow single dimensional y, we could flatten the prediction
+                Y_pred_diff = (Y_pred[..., t] - Y_pred[..., 0]).flatten()
+                cate_pred = self.models_cate[t - 1].predict(X).flatten()
+                score = np.average((Y_pred_diff - cate_pred)**2, weights=sample_weight, axis=0)
+                scores.append(score)
+            return np.mean(scores)
```

Comment:
Please add tests covering this logic; I'd recommend deriving them from my repro in the issue I filed.

Reply:
done